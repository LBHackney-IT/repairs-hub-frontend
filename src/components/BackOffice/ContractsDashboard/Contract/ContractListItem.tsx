import Contract from '@/root/src/models/contract'
import ContractExpiryDisplay from './ContractExpiryDisplay'

interface ContractListItemProps {
  contract: Contract
  index: number
  page: string
}

const ContractListItem = ({ contract, index, page }: ContractListItemProps) => {
  return (
    <>
      <li
        data-id={index}
        style={{
          border: '5px solid #00664F',
          borderRadius: '20px',
          boxSizing: 'border-box',
          padding: '1rem',
          display: 'flex',
          marginTop: '1rem',
          justifyContent: 'center',
        }}
      >
        <div className="contract-details">
          <h5
            style={{
              whiteSpace: 'nowrap',
            }}
          >
            {contract.contractReference}
          </h5>
          {page === 'contractor' && (
            <>
              <p>
                <span style={{ fontWeight: '600' }}>Sum of SORs:</span>{' '}
                {Intl.NumberFormat('en-UK', {
                  style: 'currency',
                  currency: 'GBP',
                  minimumFractionDigits: 2,
                  maximumFractionDigits: 2,
                }).format(contract.sorCost)}
              </p>
              <p>
                <span style={{ fontWeight: '600' }}>SOR Count:</span>{' '}
                {contract.sorCount}
              </p>
              <ContractExpiryDisplay contract={contract} page={page} />
            </>
          )}

          {page === 'dashboard' && (
            <>
              <p>{contract.contractorName}</p>
              <ContractExpiryDisplay contract={contract} page={page} />
            </>
          )}

          {page === 'sorSearch' && (
            <>
              <ContractExpiryDisplay contract={contract} page={page} />
            </>
          )}
        </div>
      </li>
    </>
  )
}

export default ContractListItem
