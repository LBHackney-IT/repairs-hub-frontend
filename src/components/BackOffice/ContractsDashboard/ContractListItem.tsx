import Link from 'next/link'
import cx from 'classnames'

import { dateToStr } from '@/root/src/utils/date'
import Contract from '@/root/src/models/contract'

interface ContractListItemProps {
  contract: Contract
  index: number
}

const ContractListItem = ({ contract, index }: ContractListItemProps) => {
  return (
    <Link href={`/backoffice/contracts/${contract.contractReference}`}>
      <li
        data-id={index}
        style={{ cursor: 'pointer' }}
        className={cx('govuk-!-margin-top-3', 'operative-work-order-list-item')}
      >
        <div className="appointment-details">
          <div>
            <h3 className="lbh-heading-h3 lbh-!-font-weight-bold govuk-!-margin-0 govuk-!-margin-bottom-2">
              Contract Reference: {`${contract.contractReference}`}
            </h3>
          </div>
          <h4 className="lbh-heading-h4 govuk-!-margin-0 govuk-!-margin-bottom-2 capitalize">
            {`Contractor: ${contract.contractorName}`}
          </h4>
          <h4 className="lbh-heading-h4 govuk-!-margin-0 govuk-!-margin-bottom-2">
            {`Contractor code: ${contract.contractorReference}`}
          </h4>
          <h4
            className={cx(
              'lbh-body govuk-!-margin-0 govuk-!-margin-bottom-2 capitalize'
            )}
          >
            {`Contract start date: ${dateToStr(contract.effectiveDate)}`}
          </h4>
          <h4
            className={cx(
              'lbh-body govuk-!-margin-0 govuk-!-margin-bottom-2 capitalize'
            )}
          >
            {`Contract expiry date: ${dateToStr(contract.terminationDate)}`}
          </h4>
          <h4
            className={cx(
              'lbh-body govuk-!-margin-0 govuk-!-margin-bottom-2 capitalize'
            )}
          >
            {`Count of SORs: ${contract.sorCount}`}
          </h4>
          <h4
            className={cx(
              'lbh-body govuk-!-margin-0 govuk-!-margin-bottom-2 capitalize'
            )}
          >
            {`Sum of SORs ${contract.sorCost}`}
          </h4>
        </div>
        <div className="govuk-!-margin-0">
          <span className="arrow right"></span>
        </div>
      </li>
    </Link>
  )
}

export default ContractListItem
