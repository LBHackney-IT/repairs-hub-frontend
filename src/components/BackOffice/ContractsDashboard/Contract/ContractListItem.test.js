import { render } from '@testing-library/react'
import MockDate from 'mockdate'

import ContractListItem from './ContractListItem'

import {
  mockActiveContracts,
  mockInactiveContracts,
} from '../mockContractsData'

describe('Contractors list item component', () => {
  beforeAll(() => {
    MockDate.set('2025-07-09')
  })

  afterAll(() => {
    MockDate.reset()
  })

  describe('Contractor page', () => {
    it('should render a contract with relevant fields', async () => {
      const mockContract = mockActiveContracts[0]
      const { asFragment } = render(
        <ContractListItem contract={mockContract} index={1} page="contractor" />
      )
      expect(asFragment()).toMatchSnapshot()
    })
  })
  describe('Dashboard page', () => {
    it('should render a contract that expires within two months with relevant fields', () => {
      const mockContract = {
        contractReference: '127-127-1277',
        terminationDate: '2025-08-09T15:38:48.061Z',
        effectiveDate: '2023-09-15T23:00:00Z',
        contractorReference: 'SYC',
        contractorName: 'Sycous Limited',
        isRaisable: true,
        sorCount: 0,
        sorCost: 0,
      }
      const { asFragment } = render(
        <ContractListItem contract={mockContract} index={1} page="dashboard" />
      )
      expect(asFragment()).toMatchSnapshot()
    })
  })
  describe('sorSearch', () => {
    it('should render a contract that has the sor code in it with relevant fields', () => {
      const mockContract = mockInactiveContracts[4]
      const { asFragment } = render(
        <ContractListItem contract={mockContract} index={1} page="sorSearch" />
      )
      expect(asFragment()).toMatchSnapshot()
    })
  })
})
