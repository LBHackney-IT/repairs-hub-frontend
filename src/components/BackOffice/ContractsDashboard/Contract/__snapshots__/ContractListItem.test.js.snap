// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contractors list item component Contractor page should render a contract with relevant fields 1`] = `
<DocumentFragment>
  <li
    data-id="1"
    style="border: 5px solid #00664f; border-radius: 20px; box-sizing: border-box; padding: 1rem; display: flex; margin-top: 1rem; justify-content: center;"
  >
    <div
      class="contract-details"
    >
      <h5
        style="white-space: nowrap;"
      >
        126-126-1266
      </h5>
      <p>
        <span
          style="font-weight: 600;"
        >
          Sum of SORs:
        </span>
         Â£0.00
      </p>
      <p>
        <span
          style="font-weight: 600;"
        >
          SOR Count:
        </span>
         0
      </p>
      <p>
        <span
          style="font-weight: 600;"
        >
          Expires: 
        </span>
        8 Aug 2025
      </p>
    </div>
  </li>
</DocumentFragment>
`;

exports[`Contractors list item component Dashboard page should render a contract that expires within two months with relevant fields 1`] = `
<DocumentFragment>
  <li
    data-id="1"
    style="border: 5px solid #00664f; border-radius: 20px; box-sizing: border-box; padding: 1rem; display: flex; margin-top: 1rem; justify-content: center;"
  >
    <div
      class="contract-details"
    >
      <h5
        style="white-space: nowrap;"
      >
        127-127-1277
      </h5>
      <p>
        Sycous Limited
      </p>
      <p>
        Expires: 
        <span
          style="font-weight: 600;"
        >
          8 Aug 2025
        </span>
      </p>
    </div>
  </li>
</DocumentFragment>
`;

exports[`Contractors list item component sorSearch should render a contract that has the sor code in it with relevant fields 1`] = `
<DocumentFragment>
  <li
    data-id="1"
    style="border: 5px solid #00664f; border-radius: 20px; box-sizing: border-box; padding: 1rem; display: flex; margin-top: 1rem; justify-content: center;"
  >
    <div
      class="contract-details"
    >
      <h5
        style="white-space: nowrap;"
      >
        155-155-1555
      </h5>
      <p
        style="color: red;"
      >
        Expired: 8 May 2025
      </p>
    </div>
  </li>
</DocumentFragment>
`;
